### PR TITLE
Add test cases for collection builder dialogs

### PR DESCRIPTION
**Pull request by assuntad23**
Vue Test for #12135
Vue Test for #11781 
Also, in the process of creating test, found that other PairedListCollectionCreator test was not working. Fixed that, as well. 

## How to test the changes?
(Select all options that apply)
- [X] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Review by mvdbeek — APPROVED** [review]

File: client/src/components/Collections/PairedListCollectionCreator.test.js [hunk 1]
```diff
@@ -29,6 +30,82 @@ describe("PairedListCollectionCreator", () => {
...
+                    return;
+                },
+                oncancel: () => {
+                    return;
+                },
+                hideSourceItems: false,
+            },
+        });
+        await wrapper.vm.$nextTick();
+        //change filter to .1.fastq/.2.fastq
```
Comment (jmchilton) [inline client/src/components/Collections/PairedListCollectionCreator.test.js:47]: Seems like the default filter should include the . then? Can we add a test case for the default and keep this test case for the changing the filter?

Comment (assuntad23) [reply client/src/components/Collections/PairedListCollectionCreator.test.js:47]: The "autopairs the datasets" test case uses the default filter _1/_2.

I'm not sure if we should change the default filters, it seems (or at least, my understanding is) that _1/_2 is a very common filter, as is _R1/_R2. I only added .1.fastq /.2.fastq when I vueified the collection creators.

**Comment by assuntad23** [comment]
@dannon  I fixed the client test failure, and removed the beforeEach part and added it only to the first test case. Since all three test cases use different data, that seemed the most reasonable solution.  

As for why I used a full mount instead of shallowMount. The second two (newest) test cases fail on ` [vue-test-utils]: find did not return div.forward-unpaired-filter > div.input-group-append > button, cannot call trigger() on empty Wrapper`
The first one can be a shallowMount, though, so I've converted it back.

**Comment by assuntad23** [comment]
Also, can someone re-run that API test for peace of mind? 
I don't think it's my fault that it's timing out. It is running successfully locally.

**Comment by dannon** [comment]
@assuntad23 This looks good now.  (for others -- relaying a bit of conversation from slack)  In general we want to avoid a full mount and use shallowMount as this enforces testing the component itself, and not children.  Down the road we may want to refactor this, but I think it's fine for now to just get the bugfix and new tests in.

**Pull request merged**